tests/cases/default_003.out

```diff
@@ -14,12 +14,12 @@
 ##       TYPE  VISIBILITY  ACTION      ENCODING   SIZE  NELEM  ELEMSZ  ELEMTYPE                        VALUE
 --  ---------  ----------  ------  ------------   ----  -----  ------  --------  ---------------------------
  0   REGISTER    EXPLICIT       W        OPCODE      8      1       8       INT                           bh
- 1  IMMEDIATE    EXPLICIT       R         UIMM8      8      1       8      UINT  [U A  8] 0x00000000000000A7
+ 1  IMMEDIATE    EXPLICIT       R         SIMM8      8      1       8       INT  [S A  8] 0xFFFFFFFFFFFFFFA7
 --  ---------  ----------  ------  ------------   ----  -----  ------  --------  ---------------------------
 
 == [      ATT ] ============================================================================================
-   ABSOLUTE: mov $0xA7, %bh
-   RELATIVE: mov $0xA7, %bh
+   ABSOLUTE: mov $-0x59, %bh
```

Comment:
Can we please check how other disassemblers format this case?

Reply:
NASM: `mov bh,0xa7`
Capstone: `mov bh, 0xa7`
binutils: `mov $0xa7,%bh`

In Zydis formatter properties would decide this.

Reply:
So binutils (which seems to be the only one using AT&T syntax) prints the operand as "unsigned". The problem I see here is that changing the setting in the formatter properties will change it for all signed immediate constants.

Currently it is like:

- imm -> print with sign
- uimm -> print without sign

I don't mind formatting it as "signed" for `mov`, just it seems like this is not the standard way of doing it.

Reply:
Eh, signedness is so arbitrary after all. For data transfer instructions it should be basically "no sign" and arguably "unsigned" is closer to that. However architecturally those immediates are signed, there are `mov` variants that do actual sign-extension, so to achieve consistency they must all be signed. That was one of the original arguments behind "signed by default" policy.

If we really want to preserve old behavior I would think about handling it inside formatter for reasons stated above. To be honest I don't think this new behavior is that bad at all. It's different from most, that's a downside for sure but also it's fairly common to see those one-byte regs being used to store signed stuff with -1 being probably the most common value after 0 and 1. I'd say that ignoring sign for 1-byte values is actually a popular quirk but we simply got used to it. I think I would just keep this new behavior as is.

Reply:
I agree! We just have to prepare for some issues about "wrong disassembly" I guess 😛

Reply:
There are 2 options for this already, one for immediates, second one for displacements. However they affect all immediates/displacements at once, it's a global override. Above discussion was just about default behavior when none of those overrides are active (definition signedness decides in such case).